RestaurantReservation.Db/Repositories/BaseRepository.cs

```diff
@@ -0,0 +1,7 @@
+﻿namespace RestaurantReservation.Db.Repositories
+{
+    public class BaseRepository
+    {
+        internal readonly RestaurantReservationDbContext _context = new();
+    }
+}
```

Comment:
using DI from the framework means you are NOT responsible for creating instances 
so here you are Not relying on the DI framework 
the most popular way to use DI is constructor injection 

also there are no need now to have BaseRepository and you can inject the context in each repo 

  ```
 public class BaseRepository
    {
        protected readonly RestaurantReservationDbContext _context;

        // Primary constructor with DbContext injected
        public BaseRepository(RestaurantReservationDbContext context)
        {
            _context = context;
        }
    }
}
```